common/scala/src/main/scala/whisk/core/entity/DocInfo.scala

```diff
@@ -67,17 +67,15 @@ protected[core] class DocRevision private (val rev: String) extends AnyVal {
  * @param rev the document revision, optional; this is an opaque value determined by the datastore
  */
 protected[core] case class DocInfo protected[entity] (id: DocId, rev: DocRevision = DocRevision()) {
-    def apply() = id()
-
     override def toString =
-        s"""|id: ${id()}
-            |rev: ${rev()}""".stripMargin.replace("\n", ", ")
+        s"""|id: $id
+            |rev: $rev""".stripMargin.replace("\n", ", ")
 
     override def hashCode = {
         if (rev.empty) {
             id.hashCode
         } else {
-            "$id.$rev".hashCode
+            s"$id.$rev".hashCode
```

Comment:
Would it be sensible to give a `hashCode` to `DocRevision` and leave this one be the default for `case class`es?